src/G2/Language/Monad/AST.hs

```diff
@@ -162,7 +162,12 @@ instance ASTContainerM DataCon Expr where
     modifyContainedASTsM _ dc = return dc
 
 instance ASTContainerM DataCon Type where
-    modifyContainedASTsM f (DataCon n t) = return . DataCon n =<< f t
+    modifyContainedASTsM f (DataCon n t u e) = do
+        t' <- f t 
+        u' <- traverse (modifyContainedASTsM f) u 
+        e' <- traverse (modifyContainedASTsM f) e 
```

Comment:
Don't need to use traverse here- just `modifyContainedASTsM f u`, works, for instance, because of ASTContainerM instance for lists.